source/CaptureClass.cpp

```diff
@@ -0,0 +1,107 @@
+/* CaptureClass.cpp
+Copyright (c) 2023 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "CaptureClass.h"
+
+#include "DataNode.h"
+#include "DataWriter.h"
+
+#include <algorithm>
+
+using namespace std;
+
+
+CaptureClass::CaptureClass(const DataNode &node)
+{
+	name = node.Token(1);
+
+	for(const DataNode &child : node)
+	{
+		if(child.Size() < 2)
+			child.PrintTrace("Skipping invalid capture class key with no value:");
+		const string &key = child.Token(0);
+		double value = child.Value(1);
+		if(key == "success")
```

Comment:
Since all elements are % between 1 and 0 shouldnt we have a check that the given value is in that range at the start, and a warning that it will default to 0/1 if it isnt between the two boundaries, if we dont completely disregard it?

Otherwise someone may write 50, imagining its % and get 100% applied instead (+ we're doing the same code at every line)